### PR TITLE
[process-agent/system-probe] Fix local resolution for incoming connections

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

This fixes a problem with local resolution when both ends of the connection are on the same host. Currently we end up with a network translation tacked on to both ends, even if one end is actually no getting translated. 

This PR:
* removes ip translation from a local incoming connection
* modifies the local resolution lookup to do a lookup in the network namespace first, and then a lookup without namespace
* modifies the local resolver code in process-agent to use the nat'ed raddr instead of the nat'ed laddr

### Motivation

Problems in the UI with matching both ends of a connection (i.e. by flipping source and dest, e.g.)

### Describe how to test your changes

* Run `system-probe` binary
* Run `sudo iptables -I OUTPUT -t nat -p tcp --dport 8080 -j DNAT --to-destination 127.0.0.1:8081`
* Run `nc -l -k 127.0.0.1 8081`
* In a new console/terminal, run `nc 127.0.0.1 8080 -v`. This should connect to the server that was run in the previous step. Type in something to send some data to the server. Leave this running
* For the following command you may need to replace `/var/run/sysprobe/sysprobe.sock` with the sysprobe socket path on your setup
* In another console, run `curl --unix-socket /var/run/sysprobe/sysprobe.sock http://unix/connections?client_id=3 | jq '.conns[] | select(.raddr.port==8080 and .direction=="outgoing" and .ipTranslation!=null)'`. You should see one connection
* `curl --unix-socket /var/run/sysprobe/sysprobe.sock http://unix/connections?client_id=3 | jq '.conns[] | select(.laddr.port==8081 and .direction=="incoming" and .ipTranslation==null)'`. You should see one connection again.

### CODE REVIEW

File: pkg/network/state.go [hunk 1]
```diff
@@ -663,21 +664,33 @@ func (ns *networkState) determineConnectionIntraHost(connections []ConnectionSta
...
+				_, conn.IntraHost = lAddrs[keyWithRAddr]
+			}
 		}
 
-		keyWithRAddr := newConnKey(conn, true)
-		_, ok := lAddrs[keyWithRAddr]
-		if ok {
-			conn.IntraHost = true
+		if conn.IntraHost && conn.Direction == INCOMING {
+			// remove ip translation from incoming local connections
```
Comment (leeavital) [inline pkg/network/state.go:692]: would be good to explain why here

File: pkg/network/state.go [hunk 2]
```diff
@@ -642,10 +642,11 @@ func (ns *networkState) determineConnectionIntraHost(connections []ConnectionSta
 		Address util.Address
 		Port    uint16
 		Type    ConnectionType
+		NetNS   uint32
```
[the remark this answers is not in the document]
Comment (hmahmood) [reply pkg/network/state.go:645]: For loopback addresses.

[the remark this answers is not in the document]
Comment (leeavital) [reply pkg/network/state.go:645]: That doesn't really clear it up, could you provide an example where the namespace is required?   Can we just say any connection that is to or from loopback is necessarily IntraHost ?

[the remark this answers is not in the document]
Comment (hmahmood) [reply pkg/network/state.go:645]: Yes, I think we're catching the loopback cases below anyway without the ns. I'll remove it.

**Review by leeavital — APPROVED** [review]
LGTM, though @brycekahle should take a look as well

File: pkg/process/util/netns_windows.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package util
```
Comment (shang-wang) [inline pkg/process/util/netns_windows.go:1]: Silly question, but do these files need build tags?

**Review by shang-wang — APPROVED** [review]
Just one question 👍

File: pkg/process/util/netns_windows.go (hunk 3 above)
Comment (hmahmood) [reply pkg/process/util/netns_windows.go:1]: With `_windows` filename suffix, the compiler should restrict it to windows only. This goes into more detail: https://dave.cheney.net/2013/10/12/how-to-use-conditional-compilation-with-the-go-build-tool

**Pull request merged**